action.yml

```diff
@@ -62,7 +65,8 @@ runs:
         GITHUB_SERVER_URL: ${{ inputs.github-server-url }}
         DOCKER_REGISTRY_URL: ${{ inputs.docker-registry-url }}
         DOCKER_BUILDKIT: '0' # Using remote git context with buildkit does not support referencing merge commits via SHA
-      run: docker build -t ${DOCKER_REGISTRY_URL#http*://}/$IMAGE:$TAG -f $DOCKERFILE $GITHUB_SERVER_URL/$REPOSITORY.git#$REF
+        BUILD_ARGS: ${{ inputs.build-args }}
+      run: docker build -t ${DOCKER_REGISTRY_URL#http*://}/$IMAGE:$TAG -f $DOCKERFILE ${BUILD_ARGS} $GITHUB_SERVER_URL/$REPOSITORY.git#$REF
```

Comment:
I don't think we can pass the build args via env. See how args are passed to the run command.

Reply:
https://github.com/pl-strflt/docker-container-action/blob/7058b3af9ea1d365c6f478d0ff158dd5570989fd/action.yml#L76

Reply:
I don't have a strong opinion here, but what is wrong with env variables?
I thought we wanted to use them more often than not https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#example-of-a-script-injection-attack

Reply:
Won't they get passed as one long string instead of separate args? Or does Docker parse them correctly?

What I'm getting at is that if you run something like:
```
ABC="a b c"
func numargs() { echo $#; }
func $ABC
func a b c
```

What you'll get is:
```
1
3
```

Reply:
@laurentsenta I approved, but could you make sure this is OK before merge? I think a test with 2 build args would do.

Reply:
@galargh I messed up the IFS parsing sorry about it, It's fixed now.
example run in https://github.com/laurentsenta/pl-experiments/actions/runs/5857271077/job/15878780365
You'll note the logs at the end of "build the container and run it" shows the 3 parameters as expected:

```
arg1: hello
arg2: foo
arg3: world
```

You're not going to like the why:
- bash and most shell apply word splitting on variables, which is what happen every time we don't quote a variable in CI,
- you might use zsh which does not because it was designed with arrays in mind (https://zsh.sourceforge.io/FAQ/zshfaq03.html)

```
(in zsh)
› export FILES="A B C"
› cat ${FILES}                                                  
cat: A B C: No such file or directory
› bash
bash-3.2$ cat ${FILES}
cat: A: No such file or directory
cat: B: No such file or directory
cat: C: No such file or directory
```

We're moving lists of parameters around via Github's outputs, so we'll always end up with some implicit word splitting.
I'd keep using env because at least the parameters of the step are explicit.

I think the right solution is to move around JSON and use another language, like js, to call our binaries with exactly the right argument list. I don't think we can eliminate weird word splitting, quoting, escaping, etc. if we stick to shell.